File: wci-core/src/main/kotlin/com/luggsoft/wci/core/commands/CommandRequestDescriptor.kt
package com.luggsoft.wci.core.commands

import com.fasterxml.jackson.databind.JsonNode

data class CommandRequestDescriptor(
    val title: String,
    val description: String,
    val commandRequestClass: Class<*>,
    val commandRequestSchema: JsonNode,
    val commandExecutionType: CommandExecutionType,
    val commandDefinitionOrigin: CommandDefinitionOrigin
)

File: wci-web/src/app/services/commands/command-request-descriptor.ts
import {CommandExecutionType} from "./command-execution-type";
import {CommandOriginType} from "./command-origin-type";

export interface CommandRequestDescriptor
{
    title: string;
    description: string;
    commandRequestClass: string;
    commandRequestSchema: object;
    commandExecutionType: CommandExecutionType;
    commandDefinitionOrigin: CommandOriginType;
}

